### PR TITLE
Add support for ICO

### PR DESCRIPTION
**Pull request by P1n3appl3**
This PR adds support for the [ICO format](https://en.wikipedia.org/wiki/ICO_(file_format)).

## ICO format complications

As mentioned in #8, ICO is a container format, which raises the question of which image dimensions should be retrieved. I agree with the comment there that the largest size is probably the most helpful to end users, and short of modifying the API (probably not a good idea) i think it's the best solution.

It's possible that it'd be even more helpful to return the size of the "highest quality" image contained, using some measure of size, bit depth, color planes, and encoding (ICOs can contain other formats such as PNGs), but due to complexity I think that it's fine to just take the largest. If it happens to be monochromatic, so be it. In practice I think you usually expect ICOs to just contain different scale versions of the same image so it doesn't end up mattering.

## Partial header behavior

If the ICONDIR header specifies that there will be several image entries, but we run out of bytes before we parse them all, I think we should return the largest one found so far, rather than an error. If we were reading from a file and hit EOF, then we'd want to return an `ImageError::CorruptedImage`, but if reading from a partial image file with `blob_size()` giving back a size is preferable.

## Modifying `Ord` for `ImageSize`

When writing the logic to take the largest image, I realized that `ImageSize` already derived `Ord`. While it's probably not an issue in practice (because most ICO images are squares) I think that using the area of an image would be a more natural way to order them by size. If this isn't wanted, I can revert the modified `Ord`/`PartialOrd` impls and let it take the widest rather than largest image since that's the current behavior.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -71,33 +84,33 @@ pub struct ImageSize {
...
-        } else if (header.len() >= 2 && &header[0..2] == b"\xFF\x0A")
-            || (header.len() >= 12 && &header[0..12] == b"\x00\x00\x00\x0CJXL \x0D\x0A\x87\x0A")
-        {
-            return Ok(ImageType::Jxl);
-        } else {
-            return Err(ImageError::NotSupported);
-        }
+    if header.len() < 2 {
+        Err(std::io::Error::new(std::io::ErrorKind::UnexpectedEof, "Not enough data").into())
+    } else if header.starts_with(b"\x42\x4D") {
```
Comment (P1n3appl3) [inline src/lib.rs:89]: I changed these to use [`starts_with()`](https://doc.rust-lang.org/std/primitive.slice.html#method.starts_with) which is a bit easier to read IMO. If you want I can revert it and make the minimal changes for ICO support.

Comment (Roughsketch) [reply src/lib.rs:89]: This is fine, although I do see one [0..4] for RIFF still there. Not sure if this was just skipped over or kept on purpose.

Comment (P1n3appl3) [reply src/lib.rs:89]: kept on purpose, just because it's right next to another slice comparison

Comment (P1n3appl3) [reply src/lib.rs:89]: kept on purpose, just because it's right next to another slice comparison

**Review by Roughsketch — APPROVED** [review]
Tests pass and I've been running a fuzzer on it for about a day with no issues, so seems good to me.

I will release this as 0.10.0 sometime in the next few days.

Thanks for contributing!

**Pull request merged**